js/main.js

```diff
@@ -22,22 +22,41 @@ $entryForm.addEventListener('submit', function (event) {
     note: $entryForm.notes.value,
   };
 
-  $ul.prepend(renderEntry(newEntry));
-
-  viewSwap('entries');
-  toggleNoEntries();
-
-  data.nextEntryId++;
-  data.entries.unshift(newEntry);
-  $img.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $entryForm.reset();
+  if (data.editing === null) {
+    newEntry.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    data.entries.unshift(newEntry);
+    $img.setAttribute('src', 'images/placeholder-image-square.jpg');
+    $ul.prepend(renderEntry(newEntry));
+    $entryForm.reset();
+
+    if (data.editing !== null) {
+      toggleNoEntries();
+    }
+  } else {
+    newEntry.entryId = data.editing.entryId;
+    data.entries[data.entries.length - newEntry.entryId] = newEntry;
```

Comment:
This isn't always going to work. Eventually you'll be able to delete entries, and what happens if the length is 2 but the entryId was something like 14. You'll need to find another way to select the object inside `data.entries`